packages/kbn-content-management-utils/src/msearch.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import Boom from '@hapi/boom';
+import { pick } from 'lodash';
+
+import type { StorageContext } from '@kbn/content-management-plugin/server';
+
+import type {
+  SavedObjectsFindResult,
+  SavedObject,
+  SavedObjectReference,
+} from '@kbn/core-saved-objects-api-server';
+
+import type { ServicesDefinitionSet, SOWithMetadata, SOWithMetadataPartial } from './types';
+
+type PartialSavedObject<T> = Omit<SavedObject<Partial<T>>, 'references'> & {
+  references: SavedObjectReference[] | undefined;
+};
+
+interface GetMSearchParams {
+  savedObjectType: string;
+  cmServicesDefinition: ServicesDefinitionSet;
+  allowedSavedObjectAttributes: string[];
+}
+
+function savedObjectToItem<Attributes extends object>(
```

Comment:
Just want to understand better, is this function always applicable for any saved object, or is there a scenario where it will have to be changed for a specific change object later (maybe with some data changes)?